chia/full_node/weight_proof.py

```diff
@@ -605,43 +605,43 @@ async def validate_weight_proof(self, weight_proof: WeightProof) -> Tuple[bool,
             log.error("failed weight proof sub epoch sample validation")
             return False, uint32(0), []
 
-        executor = ProcessPoolExecutor(self._num_processes)
-        constants, summary_bytes, wp_segment_bytes, wp_recent_chain_bytes = vars_to_bytes(
-            self.constants, summaries, weight_proof
-        )
-
-        recent_blocks_validation_task = asyncio.get_running_loop().run_in_executor(
-            executor, _validate_recent_blocks, constants, wp_recent_chain_bytes, summary_bytes
-        )
-
-        segments_validated, vdfs_to_validate = _validate_sub_epoch_segments(
-            constants, rng, wp_segment_bytes, summary_bytes
-        )
-        if not segments_validated:
-            return False, uint32(0), []
-
-        vdf_chunks = chunks(vdfs_to_validate, self._num_processes)
-        vdf_tasks = []
-        for chunk in vdf_chunks:
-            byte_chunks = []
-            for vdf_proof, classgroup, vdf_info in chunk:
-                byte_chunks.append((bytes(vdf_proof), bytes(classgroup), bytes(vdf_info)))
+        with ProcessPoolExecutor(self._num_processes) as executor:
```

Comment:
do we spin up these processes every time we validate a weight proof? It seems like we should just keep this `ProcessPoolExecutor` around, but maybe we only validate weight proofs very rarely.

Either way, probably not for this PR

Reply:
Yeah, I asked about this and Mariano said it was fine to just recreate the pool.